typescript/src/chain.ts

```diff
@@ -381,6 +381,40 @@ export interface TBTCVault {
   getOptimisticMintingFinalizedEvents: GetEvents.Function<OptimisticMintingFinalizedEvent>
 }
 
+/**
+ * Represnts data required to request redemption.
+ */
+export interface RequestRedemptionData {
+  /**
+   * On-chain identifier of the redeemer.
+   */
+  redeemer: Identifier
+  /**
+   * The Bitcoin public key of the wallet. Must be in the compressed form (33
+   * bytes long with 02 or 03 prefix).
+   */
+  walletPublicKey: string
+  /**
+   * The main UTXO of the wallet. Must match the main UTXO held by the on-chain
+   * Bridge contract.
+   */
+  mainUtxo: UnspentTransactionOutput
+  /**
+   * The output script that the redeemed funds will be locked to. Must be
+   * un-prefixed and not prepended with length.
+   */
+  redeemerOutputScript: string
+  /**
+   * The amount to be redeemed with the precision of the tBTC on-chain token
+   * contract.
+   */
+  amount: BigNumber
+  /**
+   * The vault address.
+   */
+  vault: Identifier
```

Comment:
Why do we need the vault as a parameter? `requestRedemption` is exposed on `TBTCToken` and there is no other choice than working with `TBTCVault` contract for this interaction.

EDIT: see the comment in `redemption.ts`.

Reply:
06601b9